guard/src/commands/github_pull.rs

```diff
@@ -0,0 +1,110 @@
+use config::Config;
+use semver::{BuildMetadata, Prerelease, Version, VersionReq};
+
+/// This is a class for getting file from GitHub
+pub struct GitHubSource {
+    pub user: String,
+    pub repo: String,
+    pub file_name: String,
+    pub access_token: String,
+    pub version_needed: String
+}
+
+
+/// Constructor
+impl GitHubSource {
+    pub fn new(user: String, repo: String, file_name: String, access_token:String, version_needed: String) -> Self {
+        GitHubSource {
+            user,
+            repo,
+            file_name,
+            access_token,
+            version_needed
+        }
+    }
+
+    /// Function to print detail of the instance
+    pub fn to_string(&self) -> String {
+        format!("GitHubSource user({}) repo ({}) file_name({}) access_token({}) version_needed({})",
+                &self.user, &self.repo, &self.file_name, &self.access_token, &self.version_needed)
+    }
+
+    /// Function to read config from TOML
+    fn read_config(&self) ->GitHubSource {
+        let settings = Config::builder()
+            // file name is "ExternalSourceConfig"
+            .add_source(config::File::with_name("src/ExternalSourceConfig"))
+            .build()
+            .unwrap();
+
+        // put details into a hashmap and create a new instance
```

Comment:
When we move this logic to a util crate, we should add multiple methods for 
1. reading from config file (parsing logic)
2. retrieving a specific parameter (this will be a wrapper on top of config library's inbuilt methods)
3. doing a sanity check for types of parameters
4. doing a sanity check around values of parameters